pkg/reconciler/apiserversource/controller_test.go

```diff
@@ -42,8 +44,13 @@ import (
 
 func TestNew(t *testing.T) {
 	ctx, _ := SetupFakeContext(t)
+
+	oidcSelector := "role=oidc-token-creator"
+	ctx = filteredFactory.WithSelectors(ctx, oidcSelector)
```

Comment:
you need to register the selector, bevor the informers are setup in `SetupFakeContext()`. You can pass some callbacks for the context, which get called, before `SetupInformers()`. See line 53-55: https://github.com/knative/eventing/blob/55c172494806f7420a4bfc913677ef3b29957336/vendor/knative.dev/pkg/reconciler/testing/context.go#L50-L60